frontend/src/Editor/LeftSidebar/index.jsx

```diff
@@ -46,10 +46,15 @@ export const LeftSidebar = forwardRef((props, ref) => {
     apps,
     dataQueries,
     clonePage,
+    queryPanelHeight,
   } = props;
   const [selectedSidebarItem, setSelectedSidebarItem] = useState();
   const [showLeaveDialog, setShowLeaveDialog] = useState(false);
   const [showDataSourceManagerModal, toggleDataSourceManagerModal] = useState(false);
+  const [popoverContentHeight, setPopoverContentHeight] = useState(queryPanelHeight);
+  useEffect(() => {
+    setPopoverContentHeight(queryPanelHeight);
```

Comment:
```suggestion
    popoverContentHeight !== queryPanelHeight && setPopoverContentHeight(queryPanelHeight);
```

Update the height only if both are not equal. It avoids unwanted renders